File: fix_ati.js
const fs = require('fs');

function fixAti() {
    let content = fs.readFileSync('c:/Users/neelg/OneDrive/Desktop/Vercel/ati.html', 'utf8');

    // 1. Remove Light/Dark mode UI and logic
    content = content.replace(/<div class="tw">[\s\S]*?<\/div>/, '');
    content = content.replace(/\.theme-switch-wrapper[\s\S]*?\{[\s\S]*?\}/, '');
    content = content.replace(/function toggleTheme[\s\S]*?\{[\s\S]*?\n\}/, '');
    content = content.replace(/function initializeTheme[\s\S]*?\{[\s\S]*?\n\}/, '');
    content = content.replace(/initializeTheme\(\);/g, '');

    // 2. Fix garbled emojis
    content = content.replace(/>[^<]*?Play Web Demo</g, '>&#128187; Play Web Demo<'); // Laptop emoji
    // Third feature card - "Active Feedback"
    content = content.replace(/<h3>[^<]*?Active Feedback<\/h3>/g, '<h3>\uD83D\uDCA1 Active Feedback</h3>'); // light bulb
    content = content.replace(/<h3>[^<]*?Progress Tracking<\/h3>/g, '<h3>\uD83D\uDCC8 Progress Tracking</h3>'); // chart
    content = content.replace(/<h3>[^<]*?Adaptive Learning<\/h3>/g, '<h3>\uD83E\uDDE0 Adaptive Learning</h3>'); // brain

    // Let's replace Mojibake  or \u0013 with nothing or space where necessary.
    content = content.replace(//g, '');

    fs.writeFileSync('c:/Users/neelg/OneDrive/Desktop/Vercel/ati.html', content);
    console.log("ati.html updated");
}
fixAti();

File: fix_ati.py
import os, re
path = os.path.join(r'c:\Users\neelg\OneDrive\Desktop\Vercel', 'ati.html')
with open(path, 'r', encoding='utf-8') as f:
    content = f.read()

# 1. Remove Light/Dark mode
content = re.sub(r'<div class="tw">.*?</div>', '', content, flags=re.DOTALL)
content = re.sub(r'\.theme-switch-wrapper\s*\{\s*display:\s*none;\s*\}', '', content, flags=re.DOTALL)
# The JS in ati.html comes from logic.js or something else. I will check logic.js later.
# For now, remove the toggle HTML.
content = re.sub(r'<div class="tw">.*?</div>', '', content, flags=re.DOTALL)

# 2. Fix Emojis
content = re.sub(r'>[^<]*? Play Web Demo<', '>\U0001F3AE Play Web Demo<', content)

# Feature cards
content = re.sub(r'<h3>[^<]*?Active Feedback</h3>', '<h3>\U0001F4A1 Active Feedback</h3>', content)
content = re.sub(r'<h3>[^<]*?Progress Tracking</h3>', '<h3>\U0001F4C8 Progress Tracking</h3>', content)
content = re.sub(r'<h3>[^<]*?Adaptive Learning</h3>', '<h3>\U0001F9E0 Adaptive Learning</h3>', content)

# Random Line bug
content = content.replace(' | ', ' &middot; ') 

# 3. Fix Images in System Interfaces
content = content.replace("'ati.png'", "'ati.webp'")
content = content.replace("'ati 2.png'", "'ati 2.webp'")
content = content.replace("'ati 3.png'", "'ati 3.webp'")

# Fix "The Sig In Problem, Some Code Error In The Site Cause Not Syncing The Data"
# It mentions "Some Code Error In The Site Cause Not Syncing The Data". This might be due to `qrs_user` vs `ati_user` in localStorage?
# Let's see later.

with open(path, 'w', encoding='utf-8') as f:
    f.write(content)
print('Fixed ati.html emojis and images and theme switch.')
